source/opaque.d.ts

```diff
@@ -162,7 +164,32 @@ const newAccountNumber = accountNumber + 2;
 
 @category Type
 */
-export type Tagged<Type, Tag extends PropertyKey> = Type & MultiTagContainer<Tag>;
+export type Tagged<Type, Tag extends PropertyKey, TagMetadata = void> = Type & MultiTagContainer<Tag, TagMetadata>;
+
+/**
+Given a type and a tag, returns the metadata associated with that tag on that type.
+
+@example
+```
```

Comment:
The example is great, but I don't think it's immediately clear to all users what problem metadata solves. Maybe explain the code example in a couple of succinct sentences?